src/quote/leveraged-extended/provider.ts

```diff
@@ -0,0 +1,217 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+import { ETH, InterestCompoundingETHIndex } from 'constants/tokens'
+import { ZeroExApi } from 'utils/0x'
+import { slippageAdjustedTokenAmount } from 'utils/slippage'
+import {
+  Exchange,
+  getSwapDataCollateralDebt,
+  getSwapDataDebtCollateral,
+  getSwapData,
+  SwapData,
+} from 'utils/swapData'
+
+import { QuoteProvider } from '../quoteProvider'
+import { QuoteToken } from '../quoteToken'
+
+import { getLeveragedTokenData } from './utils/data'
+import { getIncludedSources } from './utils/zeroex'
+
+export interface FlashMintLeveragedExtendedQuoteRequest {
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: BigNumber
```

Comment:
A change we should possibly still do is, change this to inputTokenAmount because we now wanna use the new functions (which are already used in the builder).